llvm/test/Instrumentation/RealtimeSanitizer/rtsan_unsafe.ll

```diff
@@ -0,0 +1,19 @@
+; RUN: opt < %s -passes=rtsan -S | FileCheck %s
+
+define void @_Z17blocking_functionv() #0 {
+  ret void
+}
+
+define noundef i32 @main() #2 {
+  call void @_Z17blocking_functionv() #4
+  ret i32 0
+}
+
+attributes #0 = { mustprogress noinline sanitize_realtime_unsafe optnone ssp uwtable(sync) }
```

Comment:
And just to clarify, there is no case where someone could mark something as realtime unsafe, and realtime sanitize? (this would correspond to both [[clang::nonblocking]] and [[clang::blocking]]).

This is explicitly disallowed by the attributes right? I'm basically wondering if we need a test case here to have both attributes and ensure they are inserted in the order:

realtime_enter()
notify_blocking()
realtime_exit()

not: 
notify_blocking()
realtime_enter()
realtime_exit()

Just making sure these attributes are mutually exclusive -- if they are ignore this comment completely :)

Reply:
Yeah good question - the attributes are indeed mutually exclusive. You can see the test in `main` here: https://github.com/llvm/llvm-project/blob/main/llvm/test/Verifier/rtsan-attrs.ll

Reply:
Thanks for confirming